loopy/transform/iname.py

```diff
@@ -305,7 +305,17 @@ def _split_iname_backend(kernel, iname_to_split,
     ins = _InameSplitter(rule_mapping_context, within,
             iname_to_split, outer_iname, inner_iname, new_loop_index)
 
-    kernel = ins.map_kernel(kernel)
+    from loopy.symbolic import get_dependencies, get_reduction_inames
+    from loopy.kernel.instruction import MultiAssignmentBase
+
+    def check_insn_has_iname(kernel, insn, *args):
+        return not (isinstance(insn, MultiAssignmentBase) and
```

Comment:
Why is there a `not` here?

Reply:
This should return True, if we want the instruction to be checked. If it's an MultiAssignmentBase and `iname_to_split` is not used in the assignment, we return False so that the instruction is not checked.